tests/accumulate-headers-after-304.js

```diff
@@ -0,0 +1,212 @@
+// Daft-based Squid Tests
+// Copyright (C) The Measurement Factory.
+// http://www.measurement-factory.com/
+
+// Tests whether an HTTP proxy does not allow to increase an HTTP response header
+// via 304 responses more than the specified limit.
+
+import HttpTestCase from "../src/test/HttpCase";
+import Field from "../src/http/Field";
+import Body from "../src/http/Body";
+import Resource from "../src/anyp/Resource";
+import * as ConfigurationGenerator from "../src/test/ConfigGen";
+import * as AddressPool from "../src/misc/AddressPool";
+import * as Http from "../src/http/Gadgets";
+import * as FuzzyTime from "../src/misc/FuzzyTime";
+import * as Gadgets from "../src/misc/Gadgets";
+import * as Config from "../src/misc/Config";
+import assert from "assert";
+import Test from "../src/overlord/Test";
+
+const DefaultPrefixSizeKB = 64;
+const DefaultPrefixSize = DefaultPrefixSizeKB*1024;
+
+Config.Recognize([
+    {
+        option: "smp",
+        type: "Boolean",
+        default: "false",
+        description: "In this mode MISS, UPDATE and HIT requests will go to different proxy SMP workers",
+    },
+    {
+        option: "max-prefix-size",
+        type: "Number",
+        default: DefaultPrefixSize.toString(),
+        description: "maximum message body size (KB)",
+    },
+
+]);
+
+export default class MyTest extends Test {
+
+    _configureDut(cfg) {
+        assert(!Config.dutDiskCache() || Config.smp());
+        assert(Config.dutMemoryCache() || Config.dutDiskCache());
+        if (Config.smp()) {
+            cfg.workers(4);
+            cfg.dedicatedWorkerPorts(true);
+            this._workerListeningAddresses = cfg.workerListeningAddresses();
+        } 
+        cfg.custom(`reply_header_max_size ${Config.MaxPrefixSize} KB`);
```

Comment:
Here and elsewhere in new code, please prefer Config.maxPrefixSize() style for referencing configuration options because it catches a lot more typos and stale names than Config.MaxPrefixSize style does.